### PR TITLE
Fix rfc5464 date parsing in the syslog input

### PR DESCRIPTION
**Pull request by faec**
## What does this PR do?

Fixes a [reported issue](https://github.com/elastic/beats/issues/26372) where the Syslog input misparses days of the month that begin with a `0`.

Added a test exhibiting the bug, updated the ragel spec, reran `go generate`, and confirmed that the test now passes.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] ~~I have commented my code, particularly in hard-to-understand areas~~
- [ ] ~~I have made corresponding changes to the documentation~~
- [ ] ~~I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

**Review by jsoriano — APPROVED** [review]

File: filebeat/input/syslog/format_check.go [hunk 1]
```diff
@@ -19,281 +19,252 @@
...
 
-		isRFC5424 = true
-
-		goto st7
+          isRFC5424 = true
+      
+	goto st7
 	st7:
-		if (p)++; (p) == (pe) {
+		if ( p)++; ( p) == ( pe) {
```
[the remark this answers is not in the document]
Comment (jsoriano) [reply filebeat/input/syslog/format_check.go:154]: Linters don't like them.

**Comment by jsoriano** [comment]
> make update doesn't work but make check does?

@faec what version of Go do you have locally? Check these issues https://github.com/elastic/beats/pull/26419 https://github.com/elastic/beats/pull/26186.

**Comment by faec** [comment]
> @faec what version of Go do you have locally? Check these issues #26419 #26186.

I checked the go version, it's 1.16.5 as it should be. The problem is in the build somewhere: `make update` didn't run `gofmt` on `format_check.go` so the linter failed on the whitespace. But `make check` did run it, and left the corrected file in place, so I was able to commit that.

File: filebeat/input/syslog/format_check.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ph) [reply filebeat/input/syslog/format_check.go]: This is interesting, I am not sure why Ragel would add it, I think

**Review by ph — APPROVED** [review]

File: filebeat/input/syslog/rfc5424_test.go [hunk 2]
```diff
@@ -310,6 +311,34 @@ func TestRfc5424SyslogParserValueBoundary(t *testing.T) {
 	runTests(tests, t)
 }
 
+func TestRfc5424SyslogParserDate(t *testing.T) {
+	test := []testRule{
+		{
+			title: "Test two-digit mdays",
+			log:   []byte(`<165>1 2003-08-07T05:14:15.000003-07:00 192.0.2.1 myproc 8710 - - %% It's time to make the do-nuts.`),
```
Comment (ph) [inline filebeat/input/syslog/rfc5424_test.go:318]: This test made me smile :)

**Comment by manarth** [comment]
The fix is reported in the [release notes for `7.13.3`](https://www.elastic.co/guide/en/beats/libbeat/7.13/release-notes-7.13.3.html).

The changes don't appear to be in the code tagged as [`7.13.3`](https://github.com/elastic/beats/tree/v7.13.3/filebeat/input/syslog) or [`7.13.4`](https://github.com/elastic/beats/tree/v7.13.4/filebeat/input/syslog).

Perhaps an issue with the build system, where the release notes are including references to code which isn't included in the build?

**Pull request merged**